x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,8 +14,55 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  analyticsIdIndexSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
 
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.deleteIndexPatternById(indexPatternId);
+}
+
+async function userCanDeleteIndex(
+  context: RequestHandlerContext,
+  destinationIndex: string
+): Promise<boolean> {
+  let privilege;
+  try {
+    privilege = await context.ml!.mlClient.callAsCurrentUser('transport.request', {
+      method: 'POST',
+      path: '/_security/user/_has_privileges',
+      body: {
+        index: [
+          {
+            names: [destinationIndex], // uses wildcard
+            privileges: ['delete_index'],
+          },
+        ],
+      },
+      ignoreUnavailable: true, // we allow 404 incase the user shutdown security in-between the check and now
+    });
+    return privilege.index[destinationIndex].delete_index === true;
+  } catch (err) {
+    if (
+      err.message === 'no handler found for uri [/_security/user/_has_privileges] and method [POST]'
+    ) {
+      return err;
```

Comment:
Your function should return a `Promise<boolean>`, so returning an error seems wrong 🤔

Reply:
Fixed in https://github.com/elastic/kibana/commit/b867ef2cbcad2f423d1bace93ae74ec8ca45d648